### PR TITLE
Tablebranch

### PR DESCRIPTION
**Pull request by gregoriofs**
With this query, I added additional querying functionality by using a user and the current user's friends in common to affect priority. This leads to more efficient and effective matching for users so they can partnered with the most appropriate users that are nearby. All the other criteria remain in place. As seen in the demo, the number one individual I add is the only one with whom the current user shares a friend with so they go to the top of the list, making them the most desirable friend.

My priority calculation is as follows. Generally, users who attend the same gym are given higher priority. In calculating friends in common, if the two users are members at the same gym, the value of their friendships increases by 1.5, therefore their priority increases. If a user is within a mile of a gym near the current user, then their priority increases and if that gym is their common gym, it becomes even more valuable. Then there's a final point for going to the same gym.

https://user-images.githubusercontent.com/74148230/182969658-d9e53286-2910-4ef4-abca-a88d2f3c9d52.mov

### CODE REVIEW

**Review by YitianLucyL — COMMENTED** [review]
Quick Question: why setting the 1610 miles restriction?

File: Spotti/Spotti/View Controllers/HomeViewController.m [hunk 1]
```diff
@@ -175,8 +175,8 @@ - (void)setUpStackView:(NSInteger)indexToInsert stackView:(BOOL)isFriends{
         //            UIView *view = [self makeViewforStackView:nil friendView:self.friendsAtGym[i]];
         //            [self.friendStackView addArrangedSubview:view];
         //        }
```
Comment (YitianLucyL) [inline Spotti/Spotti/View Controllers/HomeViewController.m]: remove

File: Spotti/Spotti/View Controllers/HomeViewController.m [hunk 2]
```diff
@@ -106,8 +106,8 @@ - (void)viewDidLoad {
     }];
     [self setUpStackView:1 stackView:NO];
     self.stackView.distribution = UIStackViewDistributionEqualSpacing;
-    self.stackView.spacing = 5.0;
-    self.friendStackView.spacing = 5.0;
+//    self.stackView.spacing = 5.0;
+//    self.friendStackView.spacing = 5.0;
```
Comment (YitianLucyL) [inline Spotti/Spotti/View Controllers/HomeViewController.m]: remove commented lines

File: Spotti/Spotti/View Controllers/AddingFriendsViewController.m [hunk 3]
```diff
@@ -166,23 +177,30 @@ - (void)mapSearch{
...
                         }
-                        if([item.name isEqualToString:[GymUser currentUser].gym] && ![user.gym isEqualToString:[GymUser currentUser].gym]){
-                            prios[key] = [NSNumber numberWithInt:[prios[key] intValue] + 1];
+                        else {
+                            [prios setObject:[NSNumber numberWithInt:friendsInCommon] forKey:key];
                         }
                     }
+                    if(distance <= 1610){
+                        NSInteger prio = [item.name isEqualToString:[GymUser currentUser].gym] && [user.gym isEqualToString:[GymUser currentUser].gym] ? 3 : 1;
+                        prios[key] = [NSNumber numberWithInt:[prios[key] intValue] + prio];
```
Comment (YitianLucyL) [inline Spotti/Spotti/View Controllers/AddingFriendsViewController.m:197]: Do you mind adding your priority calculation logic into the PR description?

Comment (YitianLucyL) [reply Spotti/Spotti/View Controllers/AddingFriendsViewController.m:197]: i.e. only calculate distance within 1610 miles; common friends has weight 1.5; ...

**Review by YitianLucyL — APPROVED** [review]
Overall great! Please fix nit & adding more description about your priority calculation

**Comment by gregoriofs** [comment]
My priority calculation is as follows. Generally, users who attend the same gym are given higher priority. In calculating friends in common, if the two users are members at the same gym, the value of their friendships increases by 1.5, therefore their priority increases. If a user is within a mile of a gym near the current user, then their priority increases and if that gym is their common gym, it becomes even more valuable. Then there's a final point for going to the same gym.

**Pull request merged**